src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1926,6 +1947,135 @@ namespace winrt::TerminalApp::implementation
         return true;
     }
 
+    // Detach a tree of panes from this terminal. Helper used for moving panes
+    // and tabs to other windows.
+    void TerminalPage::_DetachPaneFromWindow(std::shared_ptr<Pane> pane)
+    {
+        pane->WalkTree([&](auto p) {
+            if (const auto& control{ p->GetTerminalControl() })
+            {
+                _manager.Detach(control);
+            }
+        });
+    }
+
+    void TerminalPage::_DetachTabFromWindow(const winrt::com_ptr<TerminalTab>& terminalTab)
+    {
+        // Detach the root pane, which will act like the whole tab got detached.
+        if (const auto rootPane = terminalTab->GetRootPane())
+        {
+            _DetachPaneFromWindow(rootPane);
+        }
+    }
+
+    // Method Description:
+    // - Serialize these actions to json, and raise them as a RequestMoveContent
+    //   event. Our Window will raise that to the window manager / monarch, who
+    //   will dispatch this blob of json back to the window that should handle
+    //   this.
+    // - `actions` will be emptied into a winrt IVector as a part of this method
+    //   and should be expected to be empty after this call.
+    void TerminalPage::_MoveContent(std::vector<Settings::Model::ActionAndArgs>& actions,
```

Comment:
you might actually want to change the signature to `std::vector<>&&`. That will ensure that it can only be called with an rvalue reference, and require any caller to `move` into it.